src/core/symbology/qgssymbollayer.h

```diff
@@ -859,6 +860,14 @@ class CORE_EXPORT QgsMarkerSymbolLayer : public QgsSymbolLayer
      */
     virtual QRectF bounds( QPointF point, QgsSymbolRenderContext &context ) = 0;
 
+    /**
+     * Returns the symbol bounds of the marker symbol layer, taking into account
+     * any data defined overrides and offsets which are set for the marker layer.
+     * \returns symbol bounds, in painter units
+     * \since QGIS 3.20
+     */
+    virtual QgsMarkerSymbolBounds symbolBounds( QPointF point, QgsSymbolRenderContext &context );
```

Comment:
Why not `const QPointF &point` in this case?

Reply:
Qpointf is a small and trivially copiable class -- according to the clazy checks built into qtcreator it's quicker to copy than dereference!